### DISCUSSION TITLE
How to access the active image inside gallery and remove it from a menu button

### DISCUSSION THREAD

**Discussion by DantonGodoy** [MurhafSousli/ngx-gallery#537] [Q&A] [answered]
Hey!

Im new to your dependency and I'm trying to figure out on how to accomplish the following:

HTML:
```
<gallery
  gallerize
  #photoGallery
  [items]="galleryImages"
></gallery>
<section class="action-buttons">
  <button mat-icon-button (click)="deletePhoto(photoGallery)" [matTooltip]="'Delete active photo'">
    <mat-icon [svgIcon]="'delete'"></mat-icon>
  </button>
</section>
```

TS:
```
public deletePhoto(gallery: GalleryComponent): void {
  console.log('photo deleted.', gallery);
  gallery.remove(1); // instead of '1', should be the index from the active image
}
```

How can I access the index of the active image from the GalleryComponent API (or other API) and remove only this picture from gallery?

Great work on the project, by the way!

**Comment by MurhafSousli** [comment, marked as answer]
You can get the active index from the gallery state snapshot `gallery.galleryRef.stateSnapshot.currIndex`

**Reply by DantonGodoy** [reply]
Perfect, my friend. I was able to access the index as you told me to a little after I posted the question. Thanks a lot!
